spdx/writers/jsonyamlxml.py

```diff
@@ -116,16 +116,20 @@ def create_package_info(self, package):
         package_object["SPDXID"] = self.spdx_id(package.spdx_id)
         package_object["name"] = package.name
         package_object["downloadLocation"] = package.download_location.__str__()
-        package_object["packageVerificationCode"] = self.package_verification_code(
-            package
-        )
+        if package.files_analyzed in [True, None]:
+            package_object["packageVerificationCode"] = self.package_verification_code(
+                package
+            )
         package_object["licenseConcluded"] = self.license(package.conc_lics)
         package_object["licenseInfoFromFiles"] = list(
             map(self.license, package.licenses_from_files)
         )
         package_object["licenseDeclared"] = self.license(package.license_declared)
         package_object["copyrightText"] = package.cr_text.__str__()
 
+        if package.has_optional_field("files_analyzed"):
```

Comment:
as per spec: 3.8.3 Cardinality: Optional, one. If omitted, the default value of true is assumed.

so imho we should really initialize this to True, an only write if it is set to False

```suggestion
        if package.files_analyzed:
```